File: Java/src/main/java/Solution001_100/Solution66.java
package Solution001_100;

import java.util.ArrayList;

/**
 * @author windows
 */
public class Solution66 {
    public int[] plusOne(int[] digits) {
        ArrayList<Integer> list = new ArrayList<>();
        int carry = 1;
        for (int i = digits.length - 1; i >= 0 ; i--) {
            int sum = digits[i] + carry;
            carry = sum / 10;
            list.add(sum % 10);
        }
        if (carry == 1){
            list.add(carry);
        }
        int[] res = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            res[i] = list.get(list.size() - i - 1);
        }
        return res;
    }



}

File: C++/src/Solution1_100/Solution66.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution66.h"

vector<int> Solution66::plusOne(vector<int> &digits) {
    for (int i = digits.size() - 1; i >= 0; i--) {
        if (digits[i] == 9){
            digits[i] = 0;
        } else{
            digits[i] = digits[i] + 1;
            return digits;
        }
    }
    digits[0] = 1;
    digits.push_back(0);
    return digits;
}
